Dockerfile

```diff
@@ -1,38 +1,29 @@
-FROM docker.io/library/ubuntu:22.04 AS os-base
+ARG TARGETPLATFORM=${TARGETPLATFORM}
+ARG BUILDPLATFORM=${BUILDPLATFORM}
 
-SHELL ["/bin/bash", "-lc"]
+FROM docker.io/library/ubuntu:22.04 AS os-base
 
+ARG TARGETPLATFORM
 ARG OS_BUILD_SEED
 
 # Set the frontend to noninteractive to prevent tzdata from hanging during install
 # and set the version of Ruby to install.
 ENV DEBIAN_FRONTEND=noninteractive
-ENV RUBY_INSTALL_VERSION=3.2.2
 ENV NODE_MAJOR=20
 
-RUN ln -fs /usr/share/zoneinfo/America/New_York /etc/localtime
-
-# Run apt update and upgrade to ensure the latest packages are installed.
-RUN apt-get update ; \
-    apt-get upgrade -y
-
-# Install core dependencies.
-RUN apt-get install -y \
-    curl \
-    wget \
-    gnupg \
-    apt-transport-https \
-    lsb-release ca-certificates \
-    git
-
-# Set up the NodeSource repository.
-RUN mkdir -p /etc/apt/keyrings; \
-    curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg; \
-    echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_${NODE_MAJOR}.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list; \
-    apt-get update
-
-# Install the rest of the dependencies required to build Mastodon.
-RUN apt-get install -y \
+# Set timezone to EST/EDT, update installed packages,
+# setup NodeSource for NodeJS, install all required packages,
+# and add the 'mastodon' user.
+RUN ln -fs /usr/share/zoneinfo/America/New_York /etc/localtime ; \
+    apt-get update ; \
+    apt-get upgrade -y ; \
+    apt-get install -y curl wget gnupg apt-transport-https lsb-release ca-certificates git ; \
```

Comment:
Should `&&` be used here instead of `;`? My understanding is that `;` will still let the next command execute even if the previous one fails, which I don't think we'd want.

Reply:
Definitely. Need to do a pass for that. I started doing it on some `RUN` statements, but haven't on others.